src/modules/rust.rs

```diff
@@ -137,12 +140,12 @@ fn find_rust_toolchain_file(context: &Context) -> Option<String> {
         channel: Option<String>,
     }
 
-    fn read_channel(path: &Path) -> Option<String> {
+    fn read_channel(path: &Path, only_toml: bool) -> Option<String> {
         let contents = fs::read_to_string(path).ok()?;
 
-        match contents.lines().count() {
-            0 => None,
-            1 => Some(contents),
+        match (contents.lines().count(), only_toml) {
+            (0, _) => None,
+            (1, false) => Some(contents),
```

Comment:
I think something like this would be more clear:
```suggestion
        match contents.lines().count() {
            0 => None,
            1 if !only_toml => Some(contents),
```

Reply:
That looks good!